test/integration/common.js

```diff
@@ -0,0 +1,201 @@
+common = {
+  
+    uid: (size = 10) => {
+      let text = '';
+      const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  
+      for (let i = 0; i < size; i += 1) {
+        text += possible.charAt(Math.floor(Math.random() * possible.length));
+      }
+  
+      return text;
+    },
+    randomString: (size = 18, prefix = '') => {
+      return `${prefix}${common.uid(size)}`;
+    },
+    randomEmailAddress: () => {
+      return `${common.randomString()}@test.com`;
+    },
+    getSingleEntity: (_id, textValue, numberValue, array) => {
+      let entity = {
+        textField: textValue || common.randomString(),
+        numberField: numberValue || numberValue === 0 ? numberValue : Math.random(),
+        arrayField: array || [common.randomString(), common.randomString()]
+      };
+      if (_id) {
+        entity._id = _id;
+      }
+      return entity;
+    },
+    createData: (collectionName, arrayOfEntities) => {
+      const networkStore = Kinvey.DataStore.collection(collectionName, Kinvey.DataStoreType.Network);
+      const syncStore = Kinvey.DataStore.collection(collectionName, Kinvey.DataStoreType.Sync);
+      return Promise.all(arrayOfEntities.map(entity => {
+        return networkStore.save(entity)
+      }))
+        .then(() => syncStore.pull())
+        .then(result => _.sortBy(common.deleteEntityMetadata(result), '_id'));
+    },
+    deleteUsers: (userIds) => {
+      return Promise.all(userIds.map(userId => {
+        return Kinvey.User.remove(userId, {
+          hard: true
+        })
+          .then(() => {
+            userIds.length = 0
```

Comment:
As the function is used to delete user ids from the createdUserIds variable, which is initialized in every suite, the array should be emptied because we have to assure that there will be no attempt to delete the same user ids in a consequent call of deleteUsers.

Reply:
I understand the array needs to be emptied, but I'm saying that it shouldn't be done in here. It is not apparent that the function will modify the array, it's an unexpected side effect.

You expect it to use it to delete entities, but not to remove them from the array. I feel whoever is calling the function should take care of emptying the array.